source/MaterialXRender/ShaderValidators/Glsl/GlslProgram.cpp

```diff
@@ -300,13 +306,26 @@ void GlslProgram::bindInputs(ViewHandlerPtr viewHandler,
     bindGeometry(geometryHandler);
     bindTextures(imageHandler);
     bindTimeAndFrame();
-    bindLighting(lightHandler);
+    bindLighting(lightHandler, imageHandler);
+
+    // Set up raster state for transparency as needed
+    if (_hwShader->hasTransparency())
+    {
+        glEnable(GL_BLEND);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    }
+    else
+    {
+        glDisable(GL_BLEND);
+    }
 }
 
 void GlslProgram::unbindInputs()
 {
     unbindTextures();
     unbindGeometry();
+
+    glDisable(GL_BLEND);
```

Comment:
Minor: Should you only do this if _hwShader->hasTransparency()? I mean you can do it either way, but I think the point is you are only doing it for this case, no?